tests/store/test_obssurface.py

```diff
@@ -523,29 +463,18 @@ def test_read_noaa_metastorepack(bucket):
 
     uuid = results["processed"]["ch4_esp_surface-flask_2_representative.nc"]["ch4"]["uuid"]
 
-    ch4_data = Datasource.load(bucket=bucket, uuid=uuid, shallow=False).data()
-
-    assert sorted(list(ch4_data.keys())) == [
-        "1993-06-17-00:12:30+00:00_1993-11-20-21:50:00+00:00",
-        "1994-01-02-22:10:00+00:00_1994-12-24-22:15:00+00:00",
-        "1995-02-06-12:00:00+00:00_1995-11-08-19:55:00+00:00",
-        "1996-01-21-22:10:00+00:00_1996-12-01-20:00:00+00:00",
-        "1997-02-12-19:00:00+00:00_1997-12-20-20:15:00+00:00",
-        "1998-01-01-23:10:00+00:00_1998-12-31-19:50:00+00:00",
-        "1999-01-14-22:15:00+00:00_1999-12-31-23:35:00+00:00",
-        "2000-03-05-00:00:00+00:00_2000-11-04-22:30:00+00:00",
-        "2001-01-05-21:45:00+00:00_2001-12-06-12:00:00+00:00",
-        "2002-01-12-12:00:00+00:00_2002-01-12-12:00:01+00:00",  # Code added 1 second to ensure is not a range of 0 seconds.
-    ]
+    ch4_datasource = Datasource(bucket=bucket, uuid=uuid)
 
-    data = ch4_data["1998-01-01-23:10:00+00:00_1998-12-31-19:50:00+00:00"]
+    assert ch4_datasource.data_keys() == ["1993-06-17-00:12:30+00:00_2002-01-12-12:00:00+00:00"]
```

Comment:
What are the data keys being stored for this (and do they not include the representative date time anymore)?

Reply:
They're used to store the daterange of the dataset as it's added to the Datasource. The daterange is given by the `get_representative_daterange_str` function.